tests/core/auth/bot_authenticator_test.py

```diff
@@ -82,3 +79,27 @@ async def test_authenticate_bot(config, mocked_api_client):
 
         assert await auth_session.session_token == "session_token"
         assert await auth_session.key_manager_token == "km_token"
+
+@pytest.mark.asyncio
+async def test_authenticate_with_private_key_content(config, mocked_api_client):
+    with patch('symphony.bdk.core.auth.bot_authenticator.create_signed_jwt', return_value='privateKey'):
```

Comment:
The logic inside `create_signed_jwt` to fetch the private key content should be put in the `BdkRsaKeyConfig` class with a specific getter so that you could test create_signed_jwt is called with the correct value.